### PR TITLE
Fix broken MSSQL test

### PR DESCRIPTION
**Pull request by jedcunningham**
This broken test was causing the next test to use the db to fail. Also,
by not ignoring exceptions here we let the failure be exposed where
its broken, not in the next test that happens to run.

This was causing `TestBaseSensor.test_fail` to fail with:

```
pyodbc.ProgrammingError: ('42000', '[42000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]The server failed to resume the transaction. Desc:3300000006. (3971) (SQLExecDirectW)')
```

### CODE REVIEW

File: tests/task/task_runner/test_standard_task_runner.py [hunk 1]
```diff
@@ -228,6 +225,8 @@ def test_on_kill(self):
         for process in processes:
             assert not psutil.pid_exists(process.pid), f"{process} is still alive"
 
+        session.close()
```
Comment (potiuk) [inline tests/task/task_runner/test_standard_task_runner.py]: Should it be in try/finally clause? Or maybe even setup/tearDown?

Comment (jedcunningham) [reply tests/task/task_runner/test_standard_task_runner.py]: I can convert this to `createSession` which would do that, but calling `close` like this is pretty common in our test suite.

Comment (potiuk) [reply tests/task/task_runner/test_standard_task_runner.py]: I see a number of similar uses (but a number of finally ones as well). Actually not having finally might be a reason why we are having side effects/flaky tests so we might want to fix them all.

Comment (potiuk) [reply tests/task/task_runner/test_standard_task_runner.py]: Actually it does call for a `with newSession() as session` ... Should we make ContextManager for that? That would avoid try/finally.

Comment (potiuk) [reply tests/task/task_runner/test_standard_task_runner.py]: Maybe simply:

```
from contextlib import closing

with closing(new Session()) as session:
     ....

```

That would avoid creating an explicit context manager.

**Review by kaxil — APPROVED** [review]

File: tests/task/task_runner/test_standard_task_runner.py (hunk 1 above)
Comment (uranusjr) [reply tests/task/task_runner/test_standard_task_runner.py]: I like how your code blocks are all in a Java-ish language 😆

File: tests/task/task_runner/test_standard_task_runner.py [hunk 2]
```diff
@@ -62,15 +62,12 @@ def logging_and_db(self):
...
         dictConfig(LOGGING_CONFIG)
         yield
         airflow_logger = logging.getLogger('airflow')
         airflow_logger.handlers = []
-        try:
-            clear_db_runs()
-        except Exception:
-            # It might happen that we lost connection to the server here so we need to ignore any errors here
-            pass
+        clear_db_runs()
```
Comment (uranusjr) [inline tests/task/task_runner/test_standard_task_runner.py:71]: Is this related? I’m suspecting the try-except is there for a reason… but we can always add it back anyway.

**Review by uranusjr — COMMENTED** [review]
+1 to `create_session()`. The bare session usages go long back and we should change those where we can.

File: tests/task/task_runner/test_standard_task_runner.py (hunk 1 above)
Comment (potiuk) [reply tests/task/task_runner/test_standard_task_runner.py]: some things are never forgotten :D.

Comment (potiuk) [reply tests/task/task_runner/test_standard_task_runner.py]: I also still remember this: `ADD X TO Y GIVING Z.` (obligatory . at the end). Guess what language it is.

Comment (potiuk) [reply tests/task/task_runner/test_standard_task_runner.py]: That was true `literate programming` :D. which is becoming fashionable again.

File: tests/task/task_runner/test_standard_task_runner.py (hunk 2 above)
Comment (jedcunningham) [reply tests/task/task_runner/test_standard_task_runner.py:71]: Kinda. This was why we saw a failure in `TestBaseSensor.test_fail`, which just happened to be the next test that used the db. I'd rather see an exception here during cleanup instead of in an unrelated test as it took me a while to track down which test was responsible.

File: tests/task/task_runner/test_standard_task_runner.py (hunk 1 above)
Comment (jedcunningham) [reply tests/task/task_runner/test_standard_task_runner.py]: Fixup commit added that uses `create_session` 👍

File: tests/task/task_runner/test_standard_task_runner.py [hunk 3]
```diff
@@ -181,40 +179,40 @@ def test_on_kill(self):
...
-        pgid = os.getpgid(runner.process.pid)
-        assert pgid > 0
-        assert pgid != os.getpgid(0), "Task should be in a different process group to us"
+            processes = list(self._procs_in_pgroup(pgid))
 
-        processes = list(self._procs_in_pgroup(pgid))
+            runner.terminate()
 
-        runner.terminate()
+            session.close()  # explicitly close as `create_session`s commit will blow up otherwise
```
Comment (potiuk) [inline tests/task/task_runner/test_standard_task_runner.py:215]: should not rollback() be better here?

**Comment by uranusjr** [comment]
Hmm one MSSQL test is failing with

```
pyodbc.Error: ('40001', '[40001] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Transaction (Process ID 56)
was deadlocked on lock resources with another process and has been chosen as the deadlock victim. Rerun the
transaction. (1205) (SQLExecDirectW)')
```

File: tests/task/task_runner/test_standard_task_runner.py (hunk 3 above)
Comment (jedcunningham) [reply tests/task/task_runner/test_standard_task_runner.py:215]: `rollback` doesn't work, `terminate` seemingly already kills the transaction.

**Review by potiuk — APPROVED** [review]

**Review by kaxil — APPROVED** [review]
